mappings/maestro_file_centric.json

```diff
@@ -0,0 +1,239 @@
+      "analyzer": {
+        "autocomplete_analyzed": {
+          "filter": ["lowercase", "edge_ngram"],
+          "tokenizer": "standard"
+        },
+        "autocomplete_prefix": {
+          "filter": ["lowercase", "edge_ngram"],
+          "tokenizer": "keyword"
+        },
+        "lowercase_keyword": {
+          "filter": ["lowercase"],
+          "tokenizer": "keyword"
+        }
+      },
+      "filter": {
+        "edge_ngram": {
+          "max_gram": 20,
+          "min_gram": 20,
+          "side": "front",
+          "type": "edge_ngram"
+        }
+      }
+    }
+  },
+  "mappings": {
+    "dynamic": false,
+    "date_detection": false,
+    "properties": {
+      "file_autocomplete": {
+        "fields": {
+          "analyzed": {
+            "analyzer": "autocomplete_prefix",
+            "search_analyzer": "lowercase_keyword",
+            "type": "text"
+          },
+          "lowercase": {
+            "analyzer": "autocomplete_prefix",
+            "type": "text"
+          },
+          "prefix": {
+            "analyzer": "autocomplete_prefix",
+            "search_analyzer": "lowercase_keyword",
+            "type": "text"
+          }
+        },
+        "type": "keyword"
+      },
+      "study_id": {
+        "type": "keyword",
+        "copy_to": ["file_autocomplete"]
+      },
+      "object_id": {
+        "type": "keyword",
+        "copy_to": ["file_autocomplete"]
+      },
+      "file_type": {
+        "type": "keyword"
+      },
+      "data_type": {
+        "type": "keyword",
+        "copy_to": ["file_autocomplete"]
+      },
+      "file_access": {
+        "type": "keyword"
+      },
+      "analysis": {
+        "properties": {
+          "analysis_id": {
+            "type": "keyword",
+            "copy_to": ["file_autocomplete"]
+          },
+          "analysis_type": {
+            "type": "keyword"
+          },
+          "analysis_version": {
+            "type": "integer"
+          },
+          "experiment": {
+            "type": "object",
+            "properties": {
+              "platform": {
+                "type": "keyword"
+              },
+              "platform_model": {
+                "type": "keyword"
+              },
+              "sequencing_date": {
+                "type": "keyword"
+              },
+              "library_name": {
+                "type": "keyword"
+              },
+              "library_strategy": {
+                "type": "keyword"
+              },
+              "sequencing_center": {
+                "type": "keyword"
+              }
+            }
+          }
+        }
+      },
+      "dynamic": {
+        "properties": {
+          "workflow_name": {
+            "type": "keyword"
+          },
+          "workflow_type": {
+            "type": "keyword"
+          },
+          "analysis_software": {
+            "type": "keyword"
+          }
+        }
+      },
+      "file": {
+        "properties": {
+          "id": {
```

Comment:
is this meant for file IDs (eg, `FI12345`) like in ICGC data portal? If we'd like to continue with this type of file ID, it's possible to populate it in SONG directly instead of generate them at index building step.

Reply:
@junjun-zhang yes, i believe in the original index that maestro was built off of this meant that type of file id here.  I dont think that we would ever assign ids during the index building step.   We have not discussed the generation of `pretty file ids` yet.  

Assuming we have that discussion, do you see any other issues here or can i resolve this?

Reply:
sure, resolving it is totally fine. We can even leave it unpopulated for the time being.